contracts/protocols/mainnet/augmented-finance/v1/helpers.sol

```diff
@@ -0,0 +1,216 @@
+// SPDX-License-Identifier: agpl-3.0
+pragma solidity ^0.8.4;
+
+import { DSMath } from "../../../../utils/dsmath.sol";
+import "./interfaces.sol";
+
+contract Helpers is DSMath {
+    /**
+     * @dev Return ethereum address
+     */
+    function getEthereumAddress() internal pure returns (address) {
+        return 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE;
+    }
+
+    /**
+     * Return Weth address
+     */
+    function getWethAddress() internal pure returns (address) {
+        return 0x3B0B4787A01591DAf1bAc616465167e9f7DCD18e;
+    }
+
+    /**
+     * @dev Get Market Access Controller address
+     */
+    function getMarketAccessController() internal pure returns (address) {
+        return 0xc6f769A0c46cFFa57d91E87ED3Bc0cd338Ce6361;
+    }
+
+    /**
+     * @dev Get Procolol Data Provider address
+     */
+    function getProtocolDataProviderAddress() internal pure returns (address) {
+        return 0xd25C4a0b0c088DC8d501e4292cF28da6829023c0;
+    }
+
+    /**
+     * @dev Get Chainlink ETH price feed address
+     */
+    function getChainlinkEthFeedAddress() internal pure returns (address) {
+        return 0x5f4eC3Df9cbd43714FE2740f5E3616155c5b8419;
+    }
+
+    struct UserTokenData {
+        uint256 priceInETH;
+        uint256 priceInUSD;
+        uint256 supplyBalance;
+        uint256 stableBorrowBalance;
+        uint256 variableBorrowBalance;
+        uint256 supplyRate;
+        uint256 stableBorrowRate;
+        uint256 userStableBorrowRate;
+        uint256 variableBorrowRate;
+        bool isCollateral;
+        TokenData data;
+    }
+
+    struct TokenData {
+        uint256 ltv;
+        uint256 threshold;
+        uint256 reserveFactor;
+        bool usageAsCollateralEnabled;
+        bool borrowEnabled;
+        bool stableBorrrowEnabled;
+        bool isActive;
+        bool isFrozen;
+        uint256 totalSupply;
+        uint256 availableLiquidity;
+        uint256 totalStableDebt;
+        uint256 totalVariableDebt;
+    }
+
+    struct UserData {
+        uint256 totalCollateralETH;
+        uint256 totalBorrowsETH;
+        uint256 availableBorrowsETH;
+        uint256 currentLiquidatonThreshold;
+        uint256 ltv;
+        uint256 healthFactor;
+        uint256 ethPriceInUsd;
+        uint256 claimableRewards;
+    }
+
+    struct TokenPrice {
+        uint256 priceInETH;
+        uint256 priceInUSD;
+    }
+
+    /**
+     * @dev Get tokens prices from Chainlink price feed
+     * @param mac market access controller contract `IMarketAccessController`
+     * @param tokens tokens addresses for getting prices
+     */
+    function getTokensPrices(IMarketAccessController mac, address[] memory tokens)
+        internal
+        view
+        returns (TokenPrice[] memory tokenPrices, uint256 ethPrice)
+    {
+        uint256[] memory prices = IPriceOracle(mac.getPriceOracle()).getAssetsPrices(tokens);
+
+        ethPrice = uint256(IChainlink(getChainlinkEthFeedAddress()).latestAnswer());
```

Comment:
weth price is also available on the price oracle - it is better to get it from there